File: cpp_solutions/1442.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        int n = arr.size();
        vector<int> prefix(n + 1, 0);
        int curr = 0;
        for (int i = 0; i < n; ++i) {
            curr ^= arr[i];
            prefix[i + 1] = curr;
        }
        int res = 0;
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                if ((prefix[j + 1] ^ prefix[i]) == 0) res += (j - i);
            }
        }
        return res;
    }
};

File: rust_solutions/1442.rs
impl Solution {
    pub fn count_triplets(arr: Vec<i32>) -> i32 {
        let n = arr.len();
        let mut prefix = vec![0; n + 1];
        let mut curr: i32 = 0;
        for i in 0..n {
            curr ^= arr[i];
            prefix[i + 1] = curr;
        }
        let mut res: i32 = 0;
        for i in 0..n {
            for j in i + 1..n {
                if (prefix[j + 1] ^ prefix[i]) == 0 {
                    res += (j - i) as i32;
                }
            }
        }
        return res;
    }
}
